src/modules/polls.ts

```diff
@@ -134,25 +133,22 @@ export const scheduleAttendancePolls = async (
 
 			if (!channel) {
 				console.error(
-					`Couldn't fetch channel ${poll.channelId} for attendance poll ${poll.id}`
+					`Couldn't fetch channel ${poll.channelId} for poll ${poll.id}`
 				);
 				return;
 			}
 
-			cron.schedule(poll.cron, async () => {
+			cron.schedule(poll.cron as string, async () => {
```

Comment:
This is fine if you're trusting nobody'll call `schedulePolls` with non-scheduled polls (i.e., those with a cron field), but if you're making that assumption you might as well use `Poll & {cron: string}` as the type for the `polls` parameter.

HOWEVER, this will break when you call the function on line L252 (and in `bot.ts`) as for some reason TS isn't smart enough to infer that the `cron` field isn't `null` even with the if statement right before that line asserting that. The fix is to use `[poll as Poll & {cron: string}]` there and something similar on the respective `bot.ts` line). This is probably too much work over semantics though.

Your call whether to go extra verbose (but safer) or just leave it as it is.

Reply:
Check follow up commit